evap/grades/views.py

```diff
@@ -41,42 +44,51 @@ def course_grade_document_count_tuples(courses: QuerySet[Course]) -> list[tuple[
 
 
 @grade_publisher_required
-def semester_view(request, semester_id):
-    semester = get_object_or_404(Semester, id=semester_id)
-    if semester.grade_documents_are_deleted:
-        raise PermissionDenied
-
-    courses = (
-        semester.courses.filter(evaluations__wait_for_grade_upload_before_publishing=True)
-        .exclude(evaluations__state=Evaluation.State.NEW)
-        .distinct()
-    )
-    courses = course_grade_document_count_tuples(courses)
+class SemesterView(DetailView):
+    template_name = "grades_semester_view.html"
+    model = Semester
+    pk_url_kwarg = "semester_id"
+
+    def get_object(self, *args, **kwargs):
+        semester = super().get_object(*args, **kwargs)
+        if semester.grade_documents_are_deleted:
+            raise PermissionDenied
+        return semester
+
+    def get_context_data(self, **kwargs):
+        courses = (
+            self.object.courses.filter(evaluations__wait_for_grade_upload_before_publishing=True)
+            .exclude(evaluations__state=Evaluation.State.NEW)
+            .distinct()
+        )
+        courses = course_grade_document_count_tuples(courses)
 
-    template_data = {
-        "semester": semester,
-        "courses": courses,
-        "disable_if_archived": "disabled" if semester.grade_documents_are_deleted else "",
-        "disable_breadcrumb_semester": True,
-    }
-    return render(request, "grades_semester_view.html", template_data)
+        return super().get_context_data(**kwargs) | {
+            "courses": courses,
+            "disable_if_archived": "disabled",
+            "disable_breadcrumb_semester": True,
+        }
 
 
 @grade_publisher_or_manager_required
-def course_view(request, course_id):
-    course = get_object_or_404(Course, id=course_id)
-    semester = course.semester
-    if semester.grade_documents_are_deleted:
-        raise PermissionDenied
-
-    template_data = {
-        "semester": semester,
-        "course": course,
-        "grade_documents": course.grade_documents.all(),
-        "disable_if_archived": "disabled" if semester.grade_documents_are_deleted else "",
-        "disable_breadcrumb_course": True,
-    }
-    return render(request, "grades_course_view.html", template_data)
+class CourseView(DetailView):
+    template_name = "grades_course_view.html"
+    model = Course
+    pk_url_kwarg = "course_id"
+
+    def get_object(self, *args, **kwargs):
+        course = super().get_object(*args, **kwargs)
+        if course.semester.grade_documents_are_deleted:
+            raise PermissionDenied
+        return course
+
+    def get_context_data(self, **kwargs):
+        return super().get_context_data(**kwargs) | {
+            "semester": self.object.semester,
+            "grade_documents": self.object.grade_documents.all(),
+            "disable_if_archived": "",
```

Comment:
Passing it as explicitly empty seems more intention-conveying to me than just leaving it out, so I'd prefer that.

I'd be fine with a comment.

I'd even say the check above is business logic, not technical necessity, so it could change, so we might even want to bring back the old expression here (`disabled if blah else blah`)